diesel/src/sqlite/connection/stmt.rs

```diff
@@ -126,33 +187,166 @@ impl Drop for Statement {
     }
 }
 
-#[allow(missing_debug_implementations)]
-pub struct StatementUse<'a> {
-    statement: MaybeCached<'a, Statement>,
-    column_names: OnceCell<Vec<*const str>>,
+struct BoundStatement<'stmt, 'query> {
```

Comment:
I suggest the following changes to this part:
https://github.com/weiznich/diesel/pull/7

Reply:
I think it's unfortunately not that easy. I will write a comment at your PR explaining the issues why we should not simplify it further.